schema/m3-region.schema.json

```diff
@@ -1,6 +1,487 @@
+                "items": {
+                  "$id": "#/definitions/logicalRequirement/items/properties/resetRoom/properties/nodes/items",
+                  "type": "integer",
+                  "title": "Node ID",
+                  "description": "The in-room ID of a node."
+                }
+              },
+              "nodesToAvoid": {
+                "$id": "#/definitions/logicalRequirement/items/properties/resetRoom/properties/nodesToAvoid",
+                "type": "array",
+                "title": "Nodes to Avoid",
+                "description": "The list of nodes that must not have been visited since entering the room in order to fulfill this logical element.",
+                "items": {
+                  "$id": "#/definitions/logicalRequirement/items/properties/resetRoom/properties/nodesToAvoid/items",
+                  "type": "integer",
+                  "title": "Node ID",
+                  "description": "The in-room ID of a node."
+                }
+              },
+              "obstaclesToAvoid": {
+                "$id": "#/definitions/logicalRequirement/items/properties/resetRoom/properties/obstaclesToAvoid",
+                "type": "array",
+                "title": "Obstacles to Avoid",
+                "description": "The list of obstacles that must not have been destroyed since entering the room in order to fulfill this logical element.",
+                "items": {
+                  "$id": "#/definitions/logicalRequirement/items/properties/resetRoom/properties/obstaclesToAvoid/items",
+                  "type": "string",
+                  "title": "Obstacle ID",
+                  "description": "The in-room ID of an obstacle."
+                }
+              },
+              "mustStayPut": {
+                "$id": "#/definitions/logicalRequirement/items/properties/resetRoom/properties/mustStayPut",
+                "type": "boolean",
+                "title": "Must Stay Put",
+                "description": "Indicates that no other node but the entry node may be visited in order to fulfill this logical element. Must not be defined when it is intended to be false."
+              }
+            }
+          }
+        }
+      }
+    },
+
+    "strat": {
+      "type": "object",
+      "required": [
+        "name",
+        "notable",
+        "requires"
+      ],
+      "properties": {
+        "name": {
+          "$id": "#/definitions/strat/items/properties/name",
+          "type": "string",
+          "title": "Strat Name",
+          "description": "A name for the strat",
+          "default": "",
+          "examples": [
+            "Base",
+            "Brinstar Reserve Hole-in-one"
+          ],
+          "pattern": "^(.*)$"
+        },
+        "notable": {
+          "$id": "#/definitions/strat/properties/notable",
+          "type": "boolean"
+        },
+        "requires": {
+          "$ref" : "#/definitions/logicalRequirements",
```

Comment:
missing `type` field right?  and the `items` that goes with it.

see obstacles:requires and obstacles:bypass too

Reply:
No, that's the beauty of #ref. It's pulling the definition of `logicalRequirements` that's written higher up. So that complex type can now be referenced wherever without duplicating the definition.

Reply:
I see it now